lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -29,355 +30,436 @@
         )
-
-        grade_results = []
-        for assignment_type, assignment_info in graded_assignments.iteritems():
-            for subsection_location in assignment_info['subsection_headers']:
+        return certificate_info
+
+    def _rows_for_users(self, context, users):
+        """
+        Returns a list of rows for the given users for this report.
+        """
+        certificate_whitelist = CertificateWhitelist.objects.filter(course_id=context.course_id, whitelist=True)
+        whitelisted_user_ids = [entry.user_id for entry in certificate_whitelist]
+        success_rows, error_rows = [], []
+        for user, course_grade, err_msg in CourseGradeFactory().iter(users, course_key=context.course_id):
+            if not course_grade:
+                # An empty gradeset means we failed to grade a student.
+                error_rows.append([user.id, user.username, err_msg])
+            else:
+                success_rows.append(
+                    [user.id, user.email, user.username] +
+                    self._user_grade_results(course_grade, context) +
+                    self._user_cohort_group_names(user, context) +
+                    self._user_experiment_group_names(user, context) +
+                    self._user_team_names(user, context) +
+                    self._user_verification_mode(user, context) +
+                    self._user_certificate_info(user, context, course_grade, whitelisted_user_ids)
+                )
+        return success_rows, error_rows
+
+
+class ProblemGradeReport(object):
+    @classmethod
+    def generate(cls, _xmodule_instance_args, _entry_id, course_id, _task_input, action_name):
+        """
+        Generate a CSV containing all students' problem grades within a given
+        `course_id`.
+        """
+        start_time = time()
+        start_date = datetime.now(UTC)
+        status_interval = 100
+        enrolled_students = CourseEnrollment.objects.users_enrolled_in(course_id)
+        task_progress = TaskProgress(action_name, enrolled_students.count(), start_time)
+
+        # This struct encapsulates both the display names of each static item in the
+        # header row as values as well as the django User field names of those items
+        # as the keys.  It is structured in this way to keep the values related.
+        header_row = OrderedDict([('id', 'Student ID'), ('email', 'Email'), ('username', 'Username')])
+
+        graded_scorable_blocks = cls._graded_scorable_blocks_to_header(course_id)
+
+        # Just generate the static fields for now.
+        rows = [list(header_row.values()) + ['Grade'] + list(chain.from_iterable(graded_scorable_blocks.values()))]
+        error_rows = [list(header_row.values()) + ['error_msg']]
+        current_step = {'step': 'Calculating Grades'}
+
+        course = get_course_by_id(course_id)
+        for student, course_grade, err_msg in CourseGradeFactory().iter(enrolled_students, course):
```

Comment:
I didn't update the `ProblemGradeReport` code to chunk/parallelize yet.  For that, I've created a separate ticket: https://openedx.atlassian.net/browse/TNL-6900

I wanted to validate/test the code with Chords, etc with the `CourseGradeReport` first.  What do you think?